timon/plugins/http_server/views.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+
+# #############################################################################
+# Copyright : (C) 2023 by MHComm. All rights reserved
+#
+# __author__ = "Quentin Laymajoux"
+# __email__ = "info@mhcomm.fr"
+#
+# Name       : timon.plugins.http_server.views
+"""
+Summary: The http server plugin's views and routes
+"""
+# #############################################################################
+import json
+import logging
+import time
+
+from quart import request
+from quart_trio import QuartTrio
+
+from timon.probes.probe_if import mk_probe
+
+logger = logging.getLogger(__name__)
+
+app = QuartTrio(__name__)
+
+KNOWN_ROUTES = {
+    "/resources/": ("(GET) returns a list of all resources and their"
+                    " availability"),
+    "/queue/": "(GET) returns the heap as a list",
+    "/queue/lenght/": "(GET) returns the lenght of the heap",
+    "/queue/probe/<probename>/": "(GET) search probe in heap",
+    "/probes/<?probename>/run/": ("(GET) force run the probename and returns "
+                                  "the result"),
+    "/rescheduler/probes/": (
+        "(POST) reschedule specified probes. request args :"
+        "{'probenames': <list of probenames to reschedule>,"
+        " 'timestamp': <optional, the timestamp of the "
+        "rescheduling>}"),
+
+}
+
+
+@app.route("/")
+async def get_index():
+    """
+    returns a list of all routes and their help text
+    """
+    return KNOWN_ROUTES
+
+
+@app.route("/resources/")
+async def get_resources():
+    """
+    returns a list of all resources and their availability
+    """
+    rsrc_infos = {}
+    rsrcs = app.tmoncfg.queue.get_resources()
+    for rsrc_name, rsrc in rsrcs.items():
+        semaph = rsrc.semaph
+        rsrc_infos[rsrc_name] = {
+            "value": semaph.value,
+        }
+    return rsrc_infos
+
+
+@app.route("/queue/")
+async def get_queue():
+    """
+    returns the probes in queue
+    """
+    queue = app.tmoncfg.get_queue()
+    return queue.as_dict()
+
+
+@app.route("/queue/lenght/")
+async def get_queue_len():
+    """
+    returns the lenght of the waiting probes
+    """
+    heap = app.tmoncfg.get_queue().heap
+    data_to_return = {
+        "heap_length": len(heap)
```

Comment:
you really love the word heap, don't  you ;-)  
`heap_length` -> `queue_length`